### ISSUE TITLE
Brave content filter doesn't work on content come from ajax request

### ISSUE THREAD

**Issue by anirugu** [brave/brave-browser#40659]
### Description

Today I found that Brave has support for content-filter.  I tried some experiment and I see it doesn't work on content coming from ajax request.

### Steps to reproduce

1. Set the content-filter
2.  Try with page where content is rendered on server. It's working in that situation.
3.  Try with page where content is generated from ajax request. much later after page load.

### Actual result

Suppose you load a web-page. if the filter has something for that content, it will work. On some website the content come after the page load. Those pages doesn't work in Brave content filter.

! 2025-04-05 https://www.abcd.com 
www.abcd.com##.jumbo-tracker:not(a:contains("blahblah"))

I was using ublock origin. In Ublock origin it's hide the div with the content. This way you can use many filter to cut out the thing that doesn't matter on page.

It's not working on Brave. Let me know if it's work some other way. If you try this thing with Ublock origin you can see it's working even the content come after some interaction like button click or content coming from ajax.

### Expected result

It should work, For example Ublock origin work on content prepared on client side, in browser.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

<html>
<body>

Brave | 1.68.134 Chromium: 127.0.6533.88 (Official Build) (64-bit)
-- | --
Revision | 71f3c9cc80f92022892bbf6a22c0450ea056ade4
OS | Windows 11 Version 24H2 (Build 26120.1350)
JavaScript | V8 12.7.224.16
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36
Command Line | "C:\Users\Ani\AppData\Local\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-domain-reliability --enable-dom-distiller --enable-distillability-service --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --lso-url=https://no-thanks.invalid --sync-url=https://sync-v2.brave.com/v2 --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater.brave.com/extensions
Executable Path | C:\Users\Ani\AppData\Local\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\Ani\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Active Variations | main@4d07ec6c7de167b65db281f3f9ba752909a4bdd8AllowCertainClientHintsStudy:EnabledBraveAIChatEnabledStudy:EnabledBraveAdblockMobileNotificationsListDefault:EnabledBraveAds.CreativeAdModelBasedPredictorRecencyStudy:NoRecencyBraveAdsAdEventStudy:EnabledBraveAdsConversionsStudy:EnabledBraveAdsExclusionRulesStudy:EnabledBraveAdsNewTabPageAdsStudy:EnabledBraveAdsSiteVisitStudy:EnabledBraveAdsTextClassificationPageProbabilitiesStudy:EnabledBraveAggressiveModeRetirementExperiment:DisabledBraveAutoTranslateStudy:AutoTranslateDisabledBraveDebounceStudy:EnabledBraveFeedUpdateStudy:EnabledBraveForgetFirstPartyStorage:EnabledBraveGoogleSignInPermissionStudy:EnabledBraveHorizontalTabsUpdateEnabledStudy:EnabledBraveP3AConstellationEnabled:EnabledBraveP3AJSONOtherDeprecation:EnabledBraveP3ANebulaRelease:DefaultBraveP3ATypicalJSONDeprecationEnabled:EnabledBraveRewardsAllowSelfCustodyProvidersStudy:EnabledBraveRewardsWebUiPanelStudy:EnabledBraveScreenFingerprintingBlockerStudy:EnabledBraveSearchAdStudy:EnabledBraveSearchPromotionBannerStudyOnStable:DefaultBraveWalletAnkrBalancesEnabled:DefaultClampPlatformVersionClientHint:DisabledCookieListDefaultStudy:EnabledCrossPlatformVPNStudy:EnabledDefault1pBlockingStudy:DisabledDisableReduceLanguage:EnabledEphemeralStorageStudy:EnabledExtensionsManifestV2StudyRelease:EnabledModuleFileNamePatchStudyRelease:DefaultPartitionedCookies:EnabledSidebarEnabledStudy:DefaultSpeedreader TTS:EnabledUndecryptablePasswords:EnabledUserActivityStudy:TriggersWhatsNewStudy:Enabled

</body>
</html>

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by rebron** [comment]
cc: @antonok-edm

**Comment by antonok-edm** [comment]
@anirugu I think this is related to the filter you're looking at - it uses the `:contains(...)` procedural selector, which is not yet supported by Brave. Procedural filtering is tracked in https://github.com/brave/brave-browser/issues/16935. We'll be merging support for it soon; see https://github.com/brave/brave-core/pull/24688.

Feel free to reopen this issue with more details if you find other cases unrelated to procedural filters though!
